### DISCUSSION TITLE
Plot transparency

### DISCUSSION THREAD

**Discussion by cgoo4** [quarto-dev/quarto-cli#3385] [Q&A] [answered]
In this example the html page is grey and I'd like the plot panel/background to show through to this page grey instead of showing a white panel. Is that possible?

````
---
title: "Transparent plot panel instead of white?"
backgroundcolor: "lightgrey"
---

```{r}
#| fig-height: 1

library(ggplot2)
library(tibble)

tibble(x = 1:2, y = 1) |>
  ggplot(aes(x, y, fill = x)) +
  geom_col() +
  theme_void() +
  theme(
    legend.position = "none",
    plot.background = element_rect(fill = "transparent"),
    panel.background = element_rect(fill = "transparent")
    )
```
````
<img width="811" alt="image" src="https://user-images.githubusercontent.com/32419319/202782246-3db451dd-8386-43ba-9f56-41e732eaabdc.png">

**Comment by baptiste** [comment, marked as answer]
It's a device parameter, and they're not all consistent with the default value, unfortunately. You can change it with

````
```{r, dev='svg', dev.args=list(bg="transparent")}
...
```
````

**Reply by mcanouil** [reply]
And as YAML:

````
```r
#| dev: svg
#| dev.args: { bg: "transparent" }
```
````

or 

````
```r
#| dev: svg
#| dev.args:
#|   bg: "transparent"
```
````

**Reply by BeneKenobi** [reply]
Beside setting these options for an individual cell, it is also possible to define them for the whole document:
```yaml
---
title: "My Document"
format: html
knitr:
  opts_chunk:
    dev: svg
    dev.args: { bg: "transparent" }
---
```

Or add shared knitr options to a project-wide `_quarto.yml` file or a project-directory scoped `_metadata.yml` file (see https://quarto.org/docs/computations/execution-options.html#knitr-options).

**Reply by r-cheologist** [reply]
When using `format: revealjs`, the following in my `_quarto.yml` does not appear to do a thing in conjunction with `ggplot2` plots:
```r
knitr:
  opts_chunk:
    dev: png
    dev.args:
      bg: transparent
```
What am I doing wrong?

**Reply by mcanouil** [reply]
Please provide a small and complete example of your issue.
Thank you.

**Reply by r-cheologist** [reply]
@mcanouil: Thank you for the gentle behind kicking. I was missing the transparency in the corresponding `ggplot2` theme... Sorry for the noise. For others that may end up here: [here](https://www.r-bloggers.com/2022/07/ggplot2-transparent-background-quick-guide/) is a corresponding link...

**Reply by mcanouil** [reply]
@r-cheologist no worries. Making a reproducible example usually helps solve the issue without external help. It's similar to the "rubber duck" thing 😂
